Marlin/src/module/tool_change.cpp

```diff
@@ -398,19 +506,6 @@
     planner.buffer_line(current_position, planner.settings.max_feedrate_mm_s[Y_AXIS], active_extruder); // move away from docked toolhead
     planner.synchronize();
 
-    // STEP 6
-
-    #if HAS_HOTEND_OFFSET
-      current_position[Z_AXIS] += hotend_offset[Z_AXIS][active_extruder] - hotend_offset[Z_AXIS][tmp_extruder];
-    #endif
-
-    #if ENABLED(DEBUG_LEVELING_FEATURE)
-      if (DEBUGGING(LEVELING)) DEBUG_POS("(6) Apply Z offset", current_position);
-    #endif
-
-    #if ENABLED(DEBUG_LEVELING_FEATURE)
-      if (DEBUGGING(LEVELING)) SERIAL_ECHOLNPGM("Toolhead change done.");
-    #endif
```

Comment:
Why is the normal solenoid-based `PARKING_EXTRUDER` code being altered in this PR?

Reply:
Same Issue as in Magnetic Parking Extruder. Z-Axis setup twice. also from caller method, because MPE was based from Parking Extruder. If wanted i'll revert this change.

Reply:
I did revert Changes not related to MPE from tool_change.cpp

Reply:
If you've found an actual bug, then of course do include a fix. I just want to be sure that the changes are intentional.